comtypes/test/test_dispifc_records.py

```diff
@@ -0,0 +1,89 @@
+# coding: utf-8
+
+import unittest
+
+from comtypes import CLSCTX_LOCAL_SERVER
+from comtypes.client import CreateObject, GetModule
+from ctypes import byref, pointer
+
+ComtypesTestLib_GUID = "07D2AEE5-1DF8-4D2C-953A-554ADFD25F99"
+ProgID = "ComtypesTest.COM.Server"
+
+try:
+    GetModule([f"{{{ComtypesTestLib_GUID}}}", 1, 0, 0])
+    import comtypes.gen.ComtypesTestLib as ComtypesTestLib
+
+    IMPORT_FAILED = False
+except (ImportError, OSError):
+    IMPORT_FAILED = True
+
+
+@unittest.skipIf(IMPORT_FAILED, "This depends on the out of process COM-server.")
+class Test(unittest.TestCase):
+    """Test dispmethods with record pointer parameters."""
+
+    EXPECTED_INITED_QUESTIONS = "The meaning of life, the universe and everything?"
+
+    def _create_dispifc(self) -> "ComtypesTestLib.IComtypesTest":
+        # Explicitely ask for the dispinterface of the component.
+        return CreateObject(
+            ProgID, clsctx=CLSCTX_LOCAL_SERVER, interface=ComtypesTestLib.IComtypesTest
```

Comment:
I don't understand the problem with using the `ProgID` literal which at least tells a little about the scope of the component but I must confess that I also never came across the `IPersist_GetClassID` interface.

Reply:
The `ProgID` constant was only referenced once in this test module, and it was not clear from the name which component's `ProgID` it was.

I thought it would be clearer if we passed the `GUID` returned from the component's `IPersist_GetClassID()`.
However, through our interactions, I realized that it is easier to understand if we pass the string literal directly, without making it a constant.

```suggestion
            "ComtypesTest.COM.Server",
            clsctx=CLSCTX_LOCAL_SERVER,
            interface=ComtypesTestLib.IComtypesTest,
```
(On the other hand, even if the library ID only appears once, it makes sense to make it a constant because it is a long string and not clear what the literal implies.)